### ISSUE TITLE
Configurable lsp location for the vscode extension

### ISSUE THREAD

**Issue by viperML** [nvarner/typst-lsp#55]
- [x] I have searched the [issues](https://github.com/nvarner/typst-lsp/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [discussions](https://github.com/nvarner/typst-lsp/discussions) and believe that my question is not already covered.

## Feature Request

Add an option to use typst-lsp from PATH (or any absolute path), instead of relying the bundled excutable.

The bundled executable will not work on every linux distro.

**Comment by viperML** [comment]
Otherwise, building a static executable (with musl) could work out

**Comment by astrale-sharp** [comment]
Or wasm so that it would work for all OSs

**Comment by MilesCranmer** [comment]
I faced a similar issue and couldn't get the extension to work in vscode on my OS. I fixed it by hacking the binary used by the extension - here's how:

1. Install the typst-lsp from source:

```bash
cargo install -git https://github.com/nvarner/typst-lsp
```

2. Find the typst-lsp location being used by vscode. Open the "Output" pane at the bottom, and go to "Typst Language Server". Mine is stored at `/mnt/home/mcranmer/.vscode/extensions/nvarner.typst-lsp-0.3.2-linux-x64/out/typst-lsp`.
3. Replace this binary with a link to the compiled one. Update `EXTENSION_TYPST_LSP` and `COMPILED_TYPST_LSP` to your system's:
```
EXTENSION_TYPST_LSP=/mnt/home/mcranmer/.vscode/extensions/nvarner.typst-lsp-0.3.2-linux-x64/out/typst-lsp
COMPILED_TYPST_LSP=$HOME/.cargo/bin/typst-lsp
mv "${EXTENSION_TYPST_LSP}" "${EXTENSION_TYPST_LSP}.bkup"
ln -s $COMPILED_TYPST_LSP $EXTENSION_TYPST_LSP
```
